### PR TITLE
FBP Internal Components

### PR DESCRIPTION
**Pull request by Mechoid**
Preformatting for my changelog, donut mind.

  - bugfix: "FBPs no longer generate infinite body temperature, as it stops increasing when they die."
  - tweak: "FBP components now generate small amounts of heat."
  - tweak: "Isolated FBP heating decreased due to component heating."
  - rscadd: "Four new FBP components."
  - rscadd: "Heart analog, the hydraulic hub. Determines efficiency of the stomach analog; Reagent cycler, which only serves to move chemicals into the bloodstream of the FBP and possibly act as another stage of 'efficiency' grading for future expansion; Heatsink, which regulates temperature, damaging it can cause overheating; Diagnostic controller, which adds or removes the self-diagnosis verb. Can be implanted into non-synths to act as a gauge for their prosthetic / assisted components. "
  - tweak: "Blood now won't mix in a beaker or other container, which previously nullified the species rejection system."
  - tweak: "Species can have unique blood names in-container, on ground."
  - tweak: "FBPs have blood names dependent on their torso company. Vey is coolant, others are Oil for now."
  - rscadd: "Reagents can process in FBPs if `affects_robots` is true. Nanites have this enabled, as does Coolant."
  - rscadd: "Coolant can be used to refill an FBP's pseudo-blood. Only truly important when the FBP is at severe 'bloodloss', due to instability."

### CODE REVIEW

**Comment by Atermonera** [comment]
Does the heatsink actually work with the suit cooler?

**Comment by Mechoid** [comment]
Yes! They both function similarly, the cooler is just more powerful.

File: code/modules/organs/internal/robotic/heatsink.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
+/obj/item/organ/internal/robotic/heatsink
+	name = "heatsink"
+	icon_state = "heatsink"
+
+	organ_tag = O_HEATSINK
+
+/obj/item/organ/internal/robotic/heatsink/handle_organ_proc_special()
+	if(owner && owner.stat != DEAD)
+		owner.bodytemperature += round(owner.robobody_count * 0.9)
```
Comment (Atermonera) [inline code/modules/organs/internal/robotic/heatsink.dm:10]: No precision intended?

File: code/modules/organs/subtypes/machine.dm [hunk 2]
```diff
@@ -22,6 +22,13 @@
 	..()
 	owner.adjust_nutrition(-rand(10 / severity, 50 / severity))
 
+/obj/item/organ/internal/cell/machine/handle_organ_proc_special()
+	..()
+	if(owner && owner.stat != DEAD)
+		owner.bodytemperature += round(owner.robobody_count * 0.5)
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/organs/subtypes/machine.dm:28]: robobody_count is the number of 'vital' external organs that are robotic. Head, Torso, and Groin are the only 3 counted.

[the remark this answers is not in the document]
Comment (Atermonera) [reply code/modules/organs/subtypes/machine.dm:28]: Precision

File: code/modules/organs/subtypes/machine.dm [hunk 3]
```diff
@@ -51,6 +58,11 @@
 /obj/item/organ/internal/mmi_holder/proc/tick_defib_timer()
 	return
 
+/obj/item/organ/internal/mmi_holder/proc/get_control_efficiency()
+	. = max(0, 1 - (round(damage / max_damage * 10) / 10))
```
Comment (Atermonera) [inline code/modules/organs/subtypes/machine.dm:62]: `round(damage / max_damage, 0.1)` skips the need for the *10/10

**Pull request merged**